### ISSUE TITLE
Ladění ../admin/uzivatel/user/

### ISSUE THREAD

**Issue by motyc** [ARUP-CAS/aiscr-webamcr#132]
- [ ] když chci odstranit uživatele, vylistují se jeho vazby a přesto jde smazat; v takovém případě by vymazání vůbec nemělo být možné (pokud je vázán na historii záznamů)
- [ ] u amcr@arup.cas.cz se při kliknutí na odstranit aplikace zasekne (time-out), protože vazeb je příliš
- [ ] nesmí jít změnit ident_cely uživatele
- [ ] nikde nevidím informaci, zda byl již potvrzen email
- [ ] chybí historie úprav účtu (časy aktivace/deaktivace, změn údajů atd.)
- [ ] zadokumentovat co znamená Is staff (a celkově fungování)

**Comment by pesikj** [comment]
@motyc 
ad 2: Je potřeba to řešit, když uživatel s vazbami stejně nepůjde smazat?

**Comment by pesikj** [comment]
@motyc 
Po potvrzení e-mailu je hodnota pole `Active` změněna na True, tj. pole je zaškrtnuté.

**Comment by pesikj** [comment]
ad 6

https://github.com/ARUP-CAS/aiscr-webamcr/wiki/Administrace

hodnota is_staff určuje, zda má uživatel přístup do administrátorského rozhraní
hodnota is_active se změní na True poté, co proběhne potvrzení e-mailové adresy

**Comment by motyc** [comment]
ad 2 - pokud to bude ošetřeno tak, že se do této situace vůbec nedostanu, není s tím problém; ono teď to myslím byly dva kroky a seklo se to již v tom prvním, kdy se ukážou konkrétní vazby uživatele, než potvrdím smazání

ad 6 - ok, díky za vysvětlení

**Comment by pesikj** [comment]
@motyc Zde si asi ne úplně rozumíme. Na editačním formuláři je tlačítko pro smazání vždy a je možné na něj kliknout a přesunout se na obrazovku s přehledem relací. V případě, že smazání není možné, chybí tlačítko na smazání ve druhém kroku. Toto je standardní chování Djanga a jeho úprava by vyžadovala zásah do zdrojového kódu frameworku a tím pádem problém s kompatibilitou u nových verzí. Navíc by takové chování nedávalo smysl, protože chování aplikace by pak bylo naprosto netransparentní. Někdy by tam tlačítko bylo a někdy ne a uživatel by nevěděl, proč tomu tak je. Takto je krásně vidět důvod, proč záznam nejde smazat.

Pokud jde o příliš dlouhý seznam závislostí, možná by to šlo řešit databázovým indexem.

**Comment by motyc** [comment]
@pesikj v první části si rozumíme. Ten problém byl v tom, že právě na příkladu účtu "amcr@arup.cas.cz" aplikace vytuhne úplně - když u tohoto záznamu kliknu na "odstranit" v editačním formuláři, nedostanu se na seznam vazeb, ale systém vyhodí "504 Gateway Time-out". Do druhého kroku se tedy vůbec nedostanu, abych se na vazby podíval. Prostě skončím chybovou hláškou. Úpravu základního chování určitě nepožaduji, jen je třeba nějak ošetřit, aby systém neházel time-outy.

Jen doplňuji, že jinak je chování zcela ok (testuji např. pro "acerevkova@mzm.cz" a "Abis13@centrum.cz"), kde jsou obě varianty v pořádku (jeden uživatel smazat jde, druhý ne).

**Comment by pesikj** [comment]
@motyc Počítám, že ten problém s vytuhnutím půjde vyřešit databázovým indexem, který zrychlí načtení dat. Nastavím index a otestují ho na tomto uživateli.

**Comment by pesikj** [comment]
@motyc Záznamů je bohužel moc, takže index nepomůže. Navrhuji řešení podmíněného skrytí tlačítka "Delete", pokud je v historii více než 1000 záznamů, tím se zabrání timeoutu.

**Comment by motyc** [comment]
@pesikj Ano, tak nějak to udělat můžeme. Čistě z uživatelského hlediska - šlo by tlačítko neodstraňovat, ale jen ho nechat neaktivní a přidat tooltip s textem, např.: "Na uživatele je vázáno více než 1000 záznamů v historii."

**Comment by motyc** [comment]
@pesikj doplněna jedna opomenutá chyba (ve formuláři i tabulce chybí pole: Jméno, Příjmení a Telefon)
